programs/vote/src/vote_state.rs

```diff
@@ -489,6 +564,55 @@ pub fn process_vote(
     vote_account.set_state(&vote_state)
 }
 
+pub fn slash_state(
+    vote_account: &mut KeyedAccount,
+    slot_hashes: &[SlotHash],
+    program_id: &Pubkey,
+    tx: &Transaction,
+) -> Result<(), InstructionError> {
+    let mut vote_state: VoteState = vote_account.state()?;
+    //verify the tx is valid
+    if !tx.verify_refs() {
+        return Err(InstructionError::CustomError(
+            VoteError::InvalidSlashTransaction as u32,
+        ));
+    }
+    //verify transaction signature
+    if tx.verify().is_err() {
+        return Err(InstructionError::CustomError(
+            VoteError::InvalidSlashTransaction as u32,
+        ));
+    }
+    let signers: HashSet<Pubkey> = tx
+        .message
+        .account_keys
+        .iter()
+        .take(tx.signatures.len())
+        .cloned()
+        .collect();
+    //find vote instruction
+    for (i, ix) in tx.message.instructions.iter().enumerate() {
+        if tx.instruction_program(i) != Some(&program_id) {
+            continue;
+        }
+        if let Ok(VoteInstruction::Vote(vote)) = limited_deserialize(&ix.data) {
+            //verify that account matches mine
+            if tx.key(i, 0) != Some(vote_account.unsigned_key()) {
```

Comment:
this implies that a vote signer is allowed to sign conflicting votes as long as they go to different vote accounts?

Reply:
yea, different lockouts right?

Reply:
true.

I can't imagine a case where a signer is working with 2 validators, or where a validator is voting in 2 accounts.